### PR TITLE
Remove C++11 only to_string() function.

### PR DESCRIPTION
**Pull request by LGM-Doyle**

### CODE REVIEW

File: UI/TechTreeWnd.cpp [hunk 1]
```diff
@@ -1029,8 +1029,8 @@ void TechTreeWnd::LayoutPanel::TechPanel::Update() {
 
     if (const Tech* tech = GetTech(m_tech_name))
         m_cost_and_duration_text = boost::io::str(FlexibleFormat(UserString("TECH_TOTAL_COST_ALT_STR"))
-            % DoubleToString(tech->ResearchCost(client_empire_id), 1, false)
-            % std::to_string((int)tech->ResearchTime(client_empire_id)));
+                                                  % DoubleToString(tech->ResearchCost(client_empire_id), 1, false)
+                                                  % boost::lexical_cast<int>((int)tech->ResearchTime(client_empire_id)));
```
Comment (geoffthemedio) [inline UI/TechTreeWnd.cpp]: lose the nested c-style (int) cast

**Comment by LGM-Doyle** [comment]
rebased

**Comment by TheSilentOne1** [comment]
Thanks for the correction.

**Comment by LGM-Doyle** [comment]
Not a problem.

It is a sign that compilers (and programmers) are defaulting to C++ 11/14.

**Pull request merged**